src/main/scala/sparkz/core/network/PeerConnectionHandler.scala

```diff
@@ -203,8 +203,8 @@ class PeerConnectionHandler(val settings: NetworkSettings,
               //peer is doing bad things, ban it
               case MaliciousBehaviorException(msg) =>
                 log.warn(s"Banning peer for malicious behaviour($msg): ${connectionId.toString}")
-                //peer will be added to the blacklist and the network controller will send CloseConnection
-                networkControllerRef ! PenalizePeer(connectionId.remoteAddress, PenaltyType.PermanentPenalty)
+                // peer will be added to the blacklist and the network controller will send CloseConnection
+                networkControllerRef ! PenalizePeer(connectionId.remoteAddress, PenaltyType.MisbehaviorPenalty)
```

Comment:
since we don't ban the peer now, and don't disconnect we need to cleanup somehow `chunksBuffer` to avoid being blocked by the same error message.